packages/core/echo/echo-db/src/automerge/automerge-db.ts

```diff
@@ -47,6 +47,9 @@ export class AutomergeDb {
 
   private _ctx = new Context();
 
+  // TODO(dmaretskyi): Refactor this.
+  public readonly openTrigger = new Trigger();
```

Comment:
```suggestion
  public readonly opened = new Trigger();
```

Reply:
This pattern looks bad. Maybe add callback `onOpen` which will accept addition of index-query-source-provider

Reply:
index-query-source provider is global and databases are created per-space